CenterForOpenScience/ember-osf-web
# [ENG-2898] Collections submission failure

- Ticket: [ENG-2898]
- Feature flag: n/a

## Purpose
- Fix bug discovered in Ember upgrade testing where submission to a collection fails
- Fix another bug where the last panel of the submission page (`Collection Metadata`) does not load properly

## Summary of Changes
- Update places where the CollectedMetadatum's Collection relationship is not properly loaded

## Side Effects
- None
## QA Notes
- Submissions should now be able to be submitted, and no errors should be in the browser console
- Sometimes the `Collection Metadata` panel would not load properly if you are on the page `http://osf.io/collections/mycollection/submit` and reload the page, and this issue has been addressed in this PR as well


[ENG-2898]: https://openscience.atlassian.net/browse/ENG-2898
